### DISCUSSION TITLE
Tooltip Props Override

### DISCUSSION THREAD

**Discussion by th3oth3rjak3** [KevinVandy/mantine-react-table#426] [Ideas]
I created a custom tooltip in my application that lets me specify the transition as well as background color for the tooltip. Everywhere else in my application I use this custom tooltip, but I can't use it in the table unless I custom implement a bunch of stuff which sort of defeats the purpose of using this awesome library! What I would like to suggest is a way to allow users to specify some tooltip props that either apply to the whole table or can be configured individually. This way the styling would at least match. Here is the definition of my tooltip for an example of how I use it. If there was a way to at least provide some of these props, then I could have a consistent style.

```tsx
import { FloatingPosition, MantineTransition, Tooltip } from "@mantine/core";
import { ReactNode } from "react";

type MyTooltipProps = {
    label: string;
    children: ReactNode
    position?: FloatingPosition
}

function MyTooltip(props: MyTooltipProps) {

    const position: FloatingPosition = props.position ?? "top";

    const transition = (position: FloatingPosition): MantineTransition => {
        if (position === "left") {
            return "slide-right";
        }

        if (position === "right") {
            return "slide-left";
        }

        if (position === "top") {
            return "slide-down";
        }

        if (position === "bottom") {
            return "slide-up";
        }

        return "fade";
    }

    return (
        <Tooltip
            label={props.label}
            openDelay={1000}
            transitionProps={{ transition: transition(position), duration: 300 }}
            color="cyan"
            position={position}
            offset={10}
        >
            {props.children}
        </Tooltip>
    );
}

export default MyTooltip;
```

**Comment by KevinVandy** [comment]
MRT is a pre built TanStack Table implementation. If you don't like the muiTableBoyCellProps components, or can't customize them enough through the MUI theme, just use TanStack Table with your own custom components.

**Reply by KevinVandy** [reply]
Future versions of MRT might have more control of all internal MUI components being used, but not until at least the next major version
